src/main/java/org/apache/mesos/hdfs/state/ClusterState.java

```diff
@@ -30,13 +31,14 @@ public static ClusterState getInstance() {
   }
 
   private ClusterState() {
-    tasks = new HashMap<>();
+    taskHostMap = new HashMap<>();
+    taskSlaveMap = new HashMap<>();
     journalNodes = new HashSet<>();
     nameNodes = new HashSet<>();
-    dfsHosts = new HashSet<>();
     journalNodeHosts = new HashSet<>();
     nameNodeHosts = new HashSet<>();
```

Comment:
We'll also need to track the ZKFC and DN hosts, to ensure we can restart them, and to ensure we don't run >1 DN on the same host